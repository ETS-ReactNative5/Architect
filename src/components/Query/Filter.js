import React from 'react';
import PropTypes from 'prop-types';
import Rules from '../Rules';

const Filter = ({ rules, join, codebook, onChange, openDialog }) => (
  <Rules
    rules={rules}
    join={join}
    onChange={onChange}
    openDialog={openDialog}
    codebook={codebook}
  />
);

Filter.propTypes = {
  onChange: PropTypes.func.isRequired,
  openDialog: PropTypes.func.isRequired,
  rules: PropTypes.array.isRequired,
  codebook: PropTypes.object.isRequired,
  join: PropTypes.string.isRequired,
};

export { Filter };

export default Filter;
