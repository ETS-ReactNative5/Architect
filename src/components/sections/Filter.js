import React from 'react';
import { Field } from 'redux-form';
import { getFieldId } from '../../utils/issues';
import { Filter as FilterQuery, withFieldConnector, withStoreConnector } from '../Query';
import Section from './Section';

const FilterField = withFieldConnector(withStoreConnector(FilterQuery));

const Filter = () => (
  <Section contentId="guidance.editor.filter">
    <div id={getFieldId('filter')} data-name="Filter text" />
    <h2>Filter <small>(optional)</small></h2>
    <p>
      You can optionally filter which nodes are shown on this stage, by creating
      one or more rules using the options below.
    </p>
    <Field
      name="filter"
      component={FilterField}
    />
  </Section>
);

export default Filter;
