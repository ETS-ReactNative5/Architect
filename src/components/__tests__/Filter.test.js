/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { Filter } from '../Filter';

const mockProps = {
  variableRegistry: {},
  variableTypes: {},
};

const makeSubject = props =>
  shallow(<Filter {...mockProps} {...props} />);

describe('<Filter />', () => {
  it('can render', () => {
    const subject = makeSubject();
    expect(subject).toMatchSnapshot();
  });

  it('onChangeJoin(value)', () => {
    const onChange = jest.fn();
    const subject = makeSubject({ onChange });
    subject.instance().onChangeJoin('foo');
    expect(onChange.mock.calls[0]).toEqual([{ join: 'foo', rules: [] }]);
  });

  it('onUpdateRule(value, id, option)', () => {
    const onChange = jest.fn();

    const subject = makeSubject({
      onChange,
      filter: {
        rules: [
          { id: 'foo', options: { baz: 'hello' } },
          { id: 'bar', options: { buzz: 'world' } },
        ],
      },
    });

    subject.instance().onUpdateRule('hi', 'foo', 'baz');

    expect(onChange.mock.calls[0]).toEqual([{
      rules: [
        { id: 'foo', options: { baz: 'hi' } },
        { id: 'bar', options: { buzz: 'world' } },
      ],
    }]);
  });

  it('onMoveRule(oldIndex, newIndex)', () => {
    const onChange = jest.fn();

    const subject = makeSubject({
      onChange,
      filter: {
        rules: [
          { id: 'foo' },
          { id: 'bar' },
          { id: 'baz' },
          { id: 'buzz' },
        ],
      },
    });

    subject.instance().onMoveRule({ oldIndex: 2, newIndex: 1 });

    expect(onChange.mock.calls[0]).toEqual([{
      rules: [
        { id: 'foo' },
        { id: 'baz' },
        { id: 'bar' },
        { id: 'buzz' },
      ],
    }]);
  });

  it('onAddRule(type)', () => {
    const onChange = jest.fn();

    const subject = makeSubject({
      onChange,
      filter: {
        rules: [
        ],
      },
    });

    subject.instance().onAddRule('foo');

    expect(
      onChange.mock.calls[0][0].rules[0],
    ).toMatchObject(
      expect.objectContaining({
        type: 'foo',
        id: expect.anything(),
      }),
    );
  });

  it('onDeteRule(id)', () => {
    const onChange = jest.fn();

    const subject = makeSubject({
      onChange,
      filter: {
        rules: [
          { id: 'foo' },
          { id: 'bar' },
        ],
      },
    });

    subject.instance().onDeleteRule('foo');

    expect(
      onChange.mock.calls[0],
    ).toEqual([{ rules: [{ id: 'bar' }] }]);
  });
});
