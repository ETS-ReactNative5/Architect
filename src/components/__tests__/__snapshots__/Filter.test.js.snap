// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> can render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterGroup
    filter={
      Object {
        "join": "",
        "rules": Array [],
      }
    }
    onChange={[Function]}
    variableRegistry={Object {}}
    variableTypes={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="filter-group__join"
        >
          <DropDown
            className={null}
            onChange={[Function]}
            options={
              Array [
                "OR",
                "AND",
              ]
            }
            placeholder="{join}"
            value=""
          />
        </div>,
        <div
          className="filter-group__rules"
        >
          <sortableList
            axis="y"
            distance={0}
            getHelperDimensions={[Function]}
            hideSortableGhost={true}
            lockAxis="y"
            lockOffset="50%"
            lockToContainerEdges={false}
            onDeleteRule={[Function]}
            onSortEnd={[Function]}
            onUpdateRule={[Function]}
            pressDelay={0}
            pressThreshold={5}
            rules={Array []}
            shouldCancelStart={[Function]}
            transitionDuration={300}
            useDragHandle={true}
            useWindowAsScrollContainer={false}
            variableRegistry={Object {}}
          />
          <div
            className="filter-group__add"
          >
            <AddButton
              onAddRule={[Function]}
            />
          </div>
        </div>,
      ],
      "className": "filter-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <DropDown
            className={null}
            onChange={[Function]}
            options={
              Array [
                "OR",
                "AND",
              ]
            }
            placeholder="{join}"
            value=""
          />,
          "className": "filter-group__join",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": null,
            "onChange": [Function],
            "options": Array [
              "OR",
              "AND",
            ],
            "placeholder": "{join}",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <sortableList
              axis="y"
              distance={0}
              getHelperDimensions={[Function]}
              hideSortableGhost={true}
              lockAxis="y"
              lockOffset="50%"
              lockToContainerEdges={false}
              onDeleteRule={[Function]}
              onSortEnd={[Function]}
              onUpdateRule={[Function]}
              pressDelay={0}
              pressThreshold={5}
              rules={Array []}
              shouldCancelStart={[Function]}
              transitionDuration={300}
              useDragHandle={true}
              useWindowAsScrollContainer={false}
              variableRegistry={Object {}}
            />,
            <div
              className="filter-group__add"
            >
              <AddButton
                onAddRule={[Function]}
              />
            </div>,
          ],
          "className": "filter-group__rules",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "axis": "y",
              "distance": 0,
              "getHelperDimensions": [Function],
              "hideSortableGhost": true,
              "lockAxis": "y",
              "lockOffset": "50%",
              "lockToContainerEdges": false,
              "onDeleteRule": [Function],
              "onSortEnd": [Function],
              "onUpdateRule": [Function],
              "pressDelay": 0,
              "pressThreshold": 5,
              "rules": Array [],
              "shouldCancelStart": [Function],
              "transitionDuration": 300,
              "useDragHandle": true,
              "useWindowAsScrollContainer": false,
              "variableRegistry": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <AddButton
                onAddRule={[Function]}
              />,
              "className": "filter-group__add",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onAddRule": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="filter-group__join"
          >
            <DropDown
              className={null}
              onChange={[Function]}
              options={
                Array [
                  "OR",
                  "AND",
                ]
              }
              placeholder="{join}"
              value=""
            />
          </div>,
          <div
            className="filter-group__rules"
          >
            <sortableList
              axis="y"
              distance={0}
              getHelperDimensions={[Function]}
              hideSortableGhost={true}
              lockAxis="y"
              lockOffset="50%"
              lockToContainerEdges={false}
              onDeleteRule={[Function]}
              onSortEnd={[Function]}
              onUpdateRule={[Function]}
              pressDelay={0}
              pressThreshold={5}
              rules={Array []}
              shouldCancelStart={[Function]}
              transitionDuration={300}
              useDragHandle={true}
              useWindowAsScrollContainer={false}
              variableRegistry={Object {}}
            />
            <div
              className="filter-group__add"
            >
              <AddButton
                onAddRule={[Function]}
              />
            </div>
          </div>,
        ],
        "className": "filter-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <DropDown
              className={null}
              onChange={[Function]}
              options={
                Array [
                  "OR",
                  "AND",
                ]
              }
              placeholder="{join}"
              value=""
            />,
            "className": "filter-group__join",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": null,
              "onChange": [Function],
              "options": Array [
                "OR",
                "AND",
              ],
              "placeholder": "{join}",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <sortableList
                axis="y"
                distance={0}
                getHelperDimensions={[Function]}
                hideSortableGhost={true}
                lockAxis="y"
                lockOffset="50%"
                lockToContainerEdges={false}
                onDeleteRule={[Function]}
                onSortEnd={[Function]}
                onUpdateRule={[Function]}
                pressDelay={0}
                pressThreshold={5}
                rules={Array []}
                shouldCancelStart={[Function]}
                transitionDuration={300}
                useDragHandle={true}
                useWindowAsScrollContainer={false}
                variableRegistry={Object {}}
              />,
              <div
                className="filter-group__add"
              >
                <AddButton
                  onAddRule={[Function]}
                />
              </div>,
            ],
            "className": "filter-group__rules",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "axis": "y",
                "distance": 0,
                "getHelperDimensions": [Function],
                "hideSortableGhost": true,
                "lockAxis": "y",
                "lockOffset": "50%",
                "lockToContainerEdges": false,
                "onDeleteRule": [Function],
                "onSortEnd": [Function],
                "onUpdateRule": [Function],
                "pressDelay": 0,
                "pressThreshold": 5,
                "rules": Array [],
                "shouldCancelStart": [Function],
                "transitionDuration": 300,
                "useDragHandle": true,
                "useWindowAsScrollContainer": false,
                "variableRegistry": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <AddButton
                  onAddRule={[Function]}
                />,
                "className": "filter-group__add",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onAddRule": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
